src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -126,7 +130,11 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
         codeWriter.emitJavadoc(javadoc);
         codeWriter.emitAnnotations(annotations, false);
         codeWriter.emitModifiers(modifiers, Util.union(implicitModifiers, kind.asMemberModifiers));
-        codeWriter.emit("$L $L", kind.name().toLowerCase(Locale.US), name);
+        if (isAnnotationDeclaration()) {
```

Comment:
Wanna just add a field on `kind` like `keyword` that has values `"class"`, `"interface"`, `"enum"` and `"@interface"` ? Then we don't need this special case here. The name().toLowerCase() trick is gross when it doesn't work universally!